feat(thirdparty): sign in with apple (#748)
teamhanko/hanko: docs/docs/guides/social/apple.mdx

---
title: Login with Apple
sidebar_label: Apple
keywords:
  - social login
  - apple
  - siwa
sidebar_custom_props:
  docCardIconName: apple-dark
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# Login with Apple

To enable Apple login for your project, you need to
- Create a `Sign In With Apple`-capable application with Apple.

> - Create and your Apple credentials and configure them with Hanko.

Reviewer: ```suggestion
- Create your Apple credentials and configure them with Hanko.
```

[Docker](https://docker.com). If you do not have Docker installed, see the
[official installation instructions](https://docs.docker.com/get-docker/).

The script requires the following information:

- `private_key`: This is the path to the key file you downloaded in the [previous step](#create-key-and-download-key-file).
  Per default the key file name should look like this: `AuthKey_XXXXXXXXXX.p8`. Because we use Docker in this case, we
  will use the path to the key file on your filesystem to mount the file into the Docker container (we'll get to this,
  see the command below).
- `team_id`: This is the Team ID your Apple Developer account is associated with.
  - See [Locate your Team ID](https://developer.apple.com/help/account/manage-your-team/locate-your-team-id/#:~:text=In%20the%20top%20navigation%20of,to%20view%20your%20Team%20ID.)
    in the official Apple documentation.
- `services_id`: This is `Services ID` you created in the [Register a Services ID](#register-a-services-id) step. You
  can always review this ID in the Apple Developer console:
  - On [Certificates, Identifiers & Profiles](https://developer.apple.com/account/resources), click `Identifiers` in the sidebar,

>   then select `Services IDs` in the dropdown in the right.

Reviewer: ```suggestion
  then select `Services IDs` from the dropdown menu on the right.
```

- `key_id`: This is the ID of the private key from the [previous step](#create-key-and-download-key-file).
  The `XXXXXXXXXX` part of the default `AuthKey_XXXXXXXXXX.p8` filename is the key ID. You can
  also review this ID in the Apple Developer console:
  - On [Certificates, Identifiers & Profiles](https://developer.apple.com/account/resources), click `Keys` in the sidebar,
  then select `Services IDs`.
  - Select your key in the list.
  - The ID can be found under `View Key Details` > `Key ID`

As previously mentioned, we will mount the `private_key` to the container by mapping the path on your filesystem to
a path on the container filesystem, e.g. `~/Downloads/AuthKey_XXXXXXXXXX.p8:/tmp/AuthKey_XXXXXXXXXX.p8`.
From the command line, run (replace `<...>` placeholders with your data):

```shell
docker run -v <key_file_on_host>:<key_file_in_container> ghcr.io/teamhanko/hanko siwa \
--private_key <key_file_in_container>  \
--team_id <your_team_id> \
--services_id <your_services_id> \
--key_id <your_key_id> > client_secret.txt
```

The command will save the client secret in a file `client_secret.txt`.

> :::caution

Reviewer 2: I think a Fire would escalated things a little.

    the URL of the page that embeds the web component.
      - _**Note**_: supports wildcard matching through globbing:
        - `https://*.example.com` will allow `https://foo.example.com` and `https://bar.example.com` to be accepted.
        - `https://foo.example.com/*` will match `https://foo.example.com/page1`
          and `https://foo.example.com/page2`.
        - `**` acts as a "super"-wildcard/match-all.
- Find and expand the `Apple` provider panel in the `Providers` list.
- Enable the provider by clicking the toggle switch.
- Enter your [`Services ID`](#register-service) and client [`Secret`](#generate-secret).
- Click `Save`.

```mdx-code-block
</TabItem>
<TabItem value="self-hosted" label="Self-hosted">
```

When self-hosting you need to adjust your backend
[configuration file](https://github.com/teamhanko/hanko/blob/main/backend/config/config.yaml). You must provide
the following data:
  - your provider [`redirect_url`](#redirect-url)
  - your `client ID`, which in this case is your [`Services ID`](#configure-service)
  - your client [`secret`](#generate-secret)

>   - an`error_redirect_url`: This is a URL in your frontend the Hanko API redirects to if an error occurs during third

Reviewer 2: ```suggestion
  - an `error_redirect_url`: This is a URL in your frontend the Hanko API redirects to if an error occurs during third
```
